### PR TITLE
UI data saved to google drive

### PR DESCRIPTION
**Pull request by Sushilji**

### CODE REVIEW

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/IGoogleUploadService.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace TestingAndCalibrationLabs.Business.Core.Interfaces
+{
+    public interface DriveDownloadFile
+    {
+        string CreateFolder(string v1, string v2);
+        void UploadFile(TestReportModel getbusinessModel);
+        void UploadFileAndSendMail(TestReportModel testReportModel);
+        // void DownloadData(TestReportModel testReportModel);
+        //void DownloadData(string abc);
+
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/IGoogleUploadService.cs:14]: remove commented code

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/ITestReportService.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Collections.Generic;
+using TestingAndCalibrationLabs.Business.Common;
+using TestingAndCalibrationLabs.Business.Core.Model;
+
+namespace TestingAndCalibrationLabs.Business.Core.Interfaces
+{
+    public interface ITestReportService
+    {
+        RequestResult<int> Add(TestReportModel testReportModel);
+        bool servives(TestReportModel testReportModel);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/ITestReportService.cs:10]: what is servives method?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/UserModel.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+
+namespace TestingAndCalibrationLabs.Business.Core.Model
+{
+    public class UserModel : Entity
+    {
+        public UserModel(string emailId)
+        {
+            Email = emailId;
+        }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/UserModel.cs:12]: why are we passing email id in constructor? It's not needed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/TestingAndCalibration/ITestReportRepository.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        int Get(TestReportModel testReportModel);
+        int Update(int id, TestReportModel testReportModel);
+        List<TestReportModel> Get();
+        TestReportModel Get(int id);
+        bool Delete(int id);
+        int InsertCollection(List<TestReportModel> testReportModels);
+        TestReportModel GetTestReport(int id);
+
+
+        //object Insert(imageVal imageVal);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/TestingAndCalibration/ITestReportRepository.cs:20]: remove commented code

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/TestingAndCalibration/TestReportRepository.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        //{
+        //    string query = @"update [TestReportModel] Set 
+        //                      JobId = @JobId,
+        //                      Email = @Email,
+        //                      Client = @Client,
+        //                      Name = @Name                                
+        //                      Where Id = @Id";
+        //    using IDbConnection db = IConnectionFactory.GetConnection;
+        //    return db.Execute(query, testReportModel);
+        //}
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/TestingAndCalibration/TestReportRepository.cs:87]: remove commented code

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/GoogleUploadService.cs [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+            }
+
+            if (!isFolderExist)
+            {
+                var request = service.Files.Create(fileMetadata);
+                request.Fields = "id";
+                //var uniqueFileName = Guid.NewGuid().ToString();
+                var file = request.Execute();
+                //var result = uniqueFileName + service.Files.Create(newFile).Execute();
+                //Console.WriteLine("Folder ID: " + uniqueFileName + file.Id);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/GoogleUploadService.cs:130]: remove commented code

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/TestReportService.cs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+        public bool servives(TestReportModel testReportModel)
+        {
+            return true;
+        }
+
+        public TestReportModel GetTestReport(TestReportModel testReportModel, int id)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/TestReportService.cs:56]: why these methods are required.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+            _testReportService = testReportService;
+        }
+
+        public object NewUIModel { get; private set; }
+        public string ImageVal { get; private set; }
+
+        [HttpGet]
+        public IActionResult Index()
+        {
+            ViewBag.IsSuccess = TempData["IsTrue"] != null ? TempData["IsTrue"] : false;
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:53]: is this temp Data required.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+                    _googleUploadService.UploadFileAndSendMail(getbusinessModel);
+                }
+                TempData["IsTrue"] = true;
+                return RedirectToAction("Index");
+            }
+            return View(testReportModel);
+        }
+
+        [HttpGet]
+        public ActionResult DataView(TestReportModel testReportModel)
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:91]: DataView method name is not meaningful. give meaningful name

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+        {
+            var emailId = string.Join(",", testReportModel.Email);
+            var services = _testReportRepository.Get();
+            List<UI.Models.TestReportModel> data = new List<UI.Models.TestReportModel>();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:95]: why the variable name is data. is it making sense?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+                    Id = item.Id,
+                    Client = item.Client,
+                    DateTime = DateTime.Now.Date,
+                    DataUrl = item.DataUrl,
+                    Name = item.Name,
+                    JobId = item.JobId,
+                    FilePath = item.FilePath,
+                    Email = item.Email 
+                });
+            }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:110]: why auto mappers are not used.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+                Id = data.Id,
+                Client = data.Client,
+                //DataUrl = data.DataUrl,
+                FilePath = data.FilePath,
+                JobId = data.JobId,
+                Name = data.Name,
+                DateTime = DateTime.Now.Date,       //Accept the Current Date only
+                Email = data.Email
+            };
+            // Downloads File from Google Drive on fileId
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:136]: remove unnecessary comment

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+
+            return RedirectToAction("DataView");
+        }
+
+        [HttpGet]
+        public ActionResult TestReportMailSend( TestReportModel testReportModel, int? Id)
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:142]: rename the action name to just sendEmail

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs (hunk 11 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:110]: Auto mapper did not changed List<> data to string by .FirstorDefault() was not allowed by IDE

so this manual mapping done

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs (hunk 12 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:136]: Removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs (hunk 13 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:142]: Renamed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs (hunk 10 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:95]: renamed "data" as "testReportData" hope it is fine

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs (hunk 9 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/TestReportController.cs:91]: Renamed as TestReportView,

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/TestReportService.cs (hunk 7 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/TestReportService.cs:56]: To display item by id from TestReportmodel , also from 53 to 56 removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/TestingAndCalibration/TestReportRepository.cs (hunk 5 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/TestingAndCalibration/TestReportRepository.cs:87]: Removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/TestingAndCalibration/ITestReportRepository.cs (hunk 4 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/TestingAndCalibration/ITestReportRepository.cs:20]: Removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/ITestReportService.cs (hunk 2 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/ITestReportService.cs:10]: Removed, it was used.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/IGoogleUploadService.cs (hunk 1 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/IGoogleUploadService.cs:14]: Removed

**Pull request closed without merging**